### PR TITLE
MR: use encryption manager for input files

### PR DESCRIPTION
**Pull request by chenjunjiedada**
This along with #1526 fixes https://github.com/apache/iceberg/issues/865.

### CODE REVIEW

File: mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java [hunk 1]
```diff
@@ -79,12 +86,36 @@ public void write(DataOutput out) throws IOException {
     byte[] data = SerializationUtil.serializeToBytes(this.task);
     out.writeInt(data.length);
     out.write(data);
+
+    byte[] ioData = SerializationUtil.serializeToBytes(io);
```
Comment (chenjunjiedada) [inline mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java:90]: @rdblue @rdsr , I saw your comments in #985 about remove customized serialization. Now the `IcebergSplit` implements `org.apache.hadoop.mapred.InputSplit` which extends writable. So I keep serialization implementation. Does this make sense to you?

File: mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java [hunk 2]
```diff
@@ -79,12 +85,36 @@ public void write(DataOutput out) throws IOException {
...
+    byte[] ioData = new byte[in.readInt()];
+    in.readFully(ioData);
+    this.io = SerializationUtil.deserializeFromBytes(ioData);
+
+    byte[] encryptionManagerData = new byte[in.readInt()];
+    in.readFully(encryptionManagerData);
+    this.encryptionManager = SerializationUtil.deserializeFromBytes(encryptionManagerData);
+  }
+
+  public FileIO fileIO() {
```
Comment (rdblue) [inline mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java:113]: This is named `io` in `Table`, so I don't think a longer name is needed here.

**Comment by rdblue** [comment]
Looks good to me. Would be nice to rename that method to fit the convention used elsewhere.

**Comment by chenjunjiedada** [comment]
@rdblue , Just added Configuration serialization.

**Comment by chenjunjiedada** [comment]
Wait, this seems to serialize the configuration in `io`.

File: mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java (hunk 1 above)
Comment (rdsr) [reply mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java:89]: I'm ok with that.

**Review by rdsr — APPROVED** [review]
+1

**Comment by rdblue** [comment]
> The HadoopFileIO contains SerailiableSupplier<Configuration>, so it should be serialized by SerializationUtil and then Writable, therefore satisfy the serialization requirement for MR framework.

Where is the `HadoopFileIO` being rebuilt with `SerializableSupplier<Configuration>`? In Spark, we have to check the `FileIO` and replace it with the one that references the `SerializableSupplier`.

**Comment by chenjunjiedada** [comment]
@rdblue , Just added this. I missed that the configuration used to build `SerialableSupplier<Configuration>` also need to be serializable.

**Comment by rdblue** [comment]
Thanks for fixing it, @chenjunjiedada! I'll merge this.

**Comment by chenjunjiedada** [comment]
Thanks for merging!

**Pull request merged**